packages/imperative/src/plugins/cmd/validate/validate.handler.ts

```diff
@@ -71,10 +75,17 @@ export default class ValidateHandler implements ICommandHandler {
           "The specified plugin '" + pluginName +
           "' has not been installed into your CLI application."
         ));
+        err = true;
       } else {
-        this.displayPluginIssues(pluginName, params.response);
+        err = this.displayPluginIssues(pluginName, params.response, failOnWarning);
       }
     }
+
+    if (err === true && params.arguments.failOnError) {
+      params.response.console.log("\n");
+      params.response.console.error(TextUtils.chalk.red("Problems detected during plugin validation. Please check above for more information."));
+      params.response.data.setExitCode(1);
```

Comment:
Should we eventually implement different exit codes, based on the type of error?
For example:
- Imperative APIs (e.g.  1 - 99)
- CLI APIs (e.g. 100 - 199)
- HTTP error codes (e.g. 300 +)

Reply:
That'd be nice, but would probably require a good bit of re-work.

Reply:
More like a lot! 😋

Reply:
https://github.com/zowe/imperative/issues/636